// @bytemd/plugin-mermaid/dist/flowDiagram-v2-4c9a7611-13b0f1e7.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const addHtmlLabel = require("./add-html-label-d720b2f3.js");
const styles4fe3d1fc = require("./styles-4fe3d1fc-ec8c7e30.js");
const mermaid_core = require("./mermaid.core-067e081c.js");
require("./layout-acdefc5c.js");
require("./isPlainObject-03efe82c.js");
require("./array-87dee35f.js");
require("./path-186271b6.js");
require("./index-f9d09cc9-3a691079.js");
require("./edges-65da65dc-d0d46691.js");
require("./svgDraw-6a237a99-9b78f7fb.js");
require("./selectAll-87e3a1ba.js");
const diagram = {
  parser: addHtmlLabel.parser$1,
  db: addHtmlLabel.flowDb,
  renderer: styles4fe3d1fc.flowRendererV2,
  styles: styles4fe3d1fc.flowStyles,
  init: (cnf) => {
    if (!cnf.flowchart) {
      cnf.flowchart = {};
    }
    cnf.flowchart.arrowMarkerAbsolute = cnf.arrowMarkerAbsolute;
    mermaid_core.setConfig({ flowchart: { arrowMarkerAbsolute: cnf.arrowMarkerAbsolute } });
    styles4fe3d1fc.flowRendererV2.setConf(cnf.flowchart);
    addHtmlLabel.flowDb.clear();
    addHtmlLabel.flowDb.setGen("gen-2");
  }
};
exports.diagram = diagram;
